modules/module/BaseImageMaskModel.py

```diff
@@ -116,24 +117,20 @@ def save_mask(self):
 class BaseImageMaskModel(metaclass=ABCMeta):
     @staticmethod
     def __get_sample_filenames(sample_dir: str, include_subdirectories: bool = False) -> list[str]:
-        def __is_supported_image_extension(filename: str) -> bool:
-            ext = os.path.splitext(filename)[1]
-            return path_util.is_supported_image_extension(ext) and '-masklabel.png' not in filename
-
-        if include_subdirectories:
-            filenames = []
-            for root, _, files in os.walk(sample_dir):
-                filenames.extend(os.path.join(root, file) for file in files if __is_supported_image_extension(file))
-        else:
-            filenames = [os.path.join(sample_dir, file) for file in os.listdir(sample_dir) if __is_supported_image_extension(file)]
+        sample_dir = pathlib.Path(sample_dir)
+
+        def __is_supported_image_extension(path: pathlib.Path) -> bool:
+            ext = path.suffix
+            return path_util.is_supported_image_extension(ext) and '-masklabel.png' not in path.name
 
-        return filenames
+        recursive_prefix = "" if not include_subdirectories else "**/"
+        return [str(name) for name in sample_dir.glob(f'{recursive_prefix}*') if __is_supported_image_extension(name)]
```

Comment:
As for the argument, it can be kept as `path` since it helps readability in code autocompletion:

```py
def __is_supported_image_extension(path: Path) -> bool:
```

I always call arguments `path` too.